src/main/java/seedu/address/websocket/NusModApi.java

```diff
@@ -0,0 +1,171 @@
+package seedu.address.websocket;
+
+import java.util.logging.Logger;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.module.AcadYear;
+import seedu.address.model.module.ModuleCode;
+import seedu.address.websocket.util.ApiQuery;
+import seedu.address.websocket.util.ParserUtil;
+import seedu.address.websocket.util.QueryResult;
+
+/**
+ * NUSMods API websocket
+ * Refer to https://api.nusmods.com/v2/#/ for more detailed information
+ */
+public class NusModApi {
+    private static final String BASE_URL = "https://api.nusmods.com/v2";
+    private static final String BASE_URL2 = "http://api.nusmods.com/v2";
```

Comment:
When is this variable used? Additionally, I think it would be preferable to stick to HTTPS connections where possible to protect user privacy.